### PR TITLE
Adds Marathon support.

### PR DESCRIPTION
**Pull request by ldez**
### What does this PR do?

Adds Marathon support.

### Motivation

To have the Marathon provider.

### More

- [x] Added/updated tests
- ~[ ] Added/updated documentation~

### Additional Notes

Co-authored-by: Julien Salleyron <julien@containo.us>

### CODE REVIEW

**Comment by dtomcej** [comment]
>Motivation
To have the Docker provider.

😆

File: provider/marathon/config_test.go [hunk 1]
```diff
@@ -0,0 +1,1587 @@
...
+							},
+							Method:         "wrr",
+							PassHostHeader: true,
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "two apps with same service name and different LB method",
```
Comment (jbdoumenjou) [inline provider/marathon/config_test.go]:
```suggestion
			desc: "two apps with same service name and different LB methods",
```

File: provider/marathon/config_test.go [hunk 2]
```diff
@@ -0,0 +1,1587 @@
...
+							},
+							Method:         "wrr",
+							PassHostHeader: true,
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "two apps with two different middleware",
```
Comment (jbdoumenjou) [inline provider/marathon/config_test.go]:
```suggestion
			desc: "two apps with two different middlewares",
```

File: provider/marathon/config_test.go [hunk 3]
```diff
@@ -0,0 +1,1587 @@
...
+							},
+							Method:         "wrr",
+							PassHostHeader: true,
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "two apps with two different router with same name",
```
Comment (jbdoumenjou) [inline provider/marathon/config_test.go]:
```suggestion
			desc: "two apps with two different routers with same name",
```

File: provider/marathon/config_test.go [hunk 4]
```diff
@@ -0,0 +1,1587 @@
...
+							},
+							Method:         "wrr",
+							PassHostHeader: true,
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "two apps with two identical router with same name",
```
Comment (jbdoumenjou) [inline provider/marathon/config_test.go]:
```suggestion
			desc: "two apps with two identical routers with same name",
```

File: provider/marathon/config_test.go [hunk 5]
```diff
@@ -0,0 +1,1587 @@
...
```
Comment (jbdoumenjou) [inline provider/marathon/config_test.go]:
```suggestion
			desc: "two apps with two identical routers with same name",
```

**Comment by kopax** [comment]
Hi all, I have tested the marathon provider and I keep having 504 error. It worked fine with docker provider but that's my requirement to go to production.

```
reverse-proxy_1  | time="2019-12-11T14:25:02Z" level=debug msg="'504 Gateway Timeout' caused by: dial tcp 212.95.74.75:31968: i/o timeout"
```

`forceTaskHostname: true` did not help.

Issue reported [here](https://github.com/containous/traefik/issues/6001). I am still googling to get more tutorial with marathon but there's really nothing.

Thanks for sharing, helping and best!

**Pull request merged**